modules/currency-l1/src/main/scala/org/tessellation/currency/l1/DataApplication.scala

```diff
@@ -46,6 +45,8 @@ object DataApplication {
     lastCurrencySnapshotStorage: LastSnapshotStorage[F, CurrencyIncrementalSnapshot, CurrencySnapshotInfo]
   ): Stream[F, Unit] = {
 
+    implicit val dataEncoder = dataApplicationService.dataEncoder
```

Comment:
Needed in multiple places I guess?

Reply:
What exactly and why?

Reply:
Ah, ok. I thought it's needed in multiple places but looks like it's not. Moved to the specific function :+1: